src/components/ChannelList/hooks/listeners/useNewMessageNotification.js

```diff
@@ -0,0 +1,27 @@
+import { useContext, useEffect } from 'react';
+import uniqBy from 'lodash/uniqBy';
+
+import { ChatContext } from '../../../../context';
+import { getChannel } from '../../utils';
+
+export const useNewMessageNotification = ({ onMessageNew, setChannels }) => {
+  const { client } = useContext(ChatContext);
+
+  useEffect(() => {
+    const handleEvent = async (e) => {
+      if (onMessageNew && typeof onMessageNew === 'function') {
+        onMessageNew(setChannels, e);
+      } else {
+        const channel = await getChannel(
+          client,
+          e.channel?.type,
+          e.channel?.id,
```

Comment:
ternaries mess with docs at all?
```suggestion
          e.channel?.type,
          e.channel?.id,
```